Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -372,4 +372,17 @@ export default function widgetBehavior(publicAPI, model) {
     model._widgetManager.enablePicking();
     model._interactor.render();
   };
+
+  publicAPI.reset = () => {
+    model.widgetState.deactivate();
+    model.widgetState.getMoveHandle().deactivate();
+
+    model.widgetState.getHandle1().setOrigin(null);
+    model.widgetState.getHandle2().setOrigin(null);
+    model.widgetState.getMoveHandle().setOrigin(null);
+    model.widgetState.getText().setOrigin(null);
+    model.widgetState.getText().setText('');
+
+    model.activeState = null;
+  };
```

Comment:
what about a more generic way by calling `generateState()` again ?

Reply:
I like this idea from a high-level point of view. Problem is, I don't think we can just do `model.widgetState = generateState()`, since users may have kept a reference to the widget state externally (e.g. `model.widgetState.onModified(...)`.

Reply:
I guess we could add a `deepCopy()` convenient function in `Widgets/Core/WidgetState/index.js`.

Reply:
I think we should work on that idea outside of this PR. `deepCopy()` does sound like a good idea, but I wonder if its potential complexity outweighs manually performing a reset.

Reply:
Good point